### PR TITLE
test: Make check_ipc tidy up after itself.

### PR DESCRIPTION
**Pull request by chrissie-c**
Signed-off-by: Christine Caulfield <ccaulfie@redhat.com>

### CODE REVIEW

File: tests/check_ipc.c [hunk 1]
```diff
@@ -265,10 +265,12 @@ s1_msg_process_fn(qb_ipcs_connection_t *c,
 		}
 
 	} else if (req_pt->id == IPC_MSG_REQ_SERVER_FAIL) {
+		qb_ipcs_destroy(s1);
 		exit(0);
 	} else if (req_pt->id == IPC_MSG_REQ_SERVER_DISCONNECT) {
 		multiple_connections = QB_FALSE;
 		qb_ipcs_disconnect(c);
+		qb_ipcs_destroy(s1);
```
Comment (jnpkrn) [inline tests/check_ipc.c]: Overall, the code logic is quite hard to follow, so I would encourage adding
comments along whenever there's something not entirely apparent.

In this very case, I have no clue why you wouldn't want to substitute
`disconnect` for `destroy`, but you put it in addition instead, why a superficial
look at the latter suggests the former is implied...

Comment (chrissie-c) [inline tests/check_ipc.c]: Thanks. testing this on FreeBSD causes breakage so I think the qb_ipcs_destroy() calls cause the sockets to be closed when the client test is expecting them to be still there. I think I might have to revert to the manual unlink that I had to use into test_ipcc_truncate_when_unlink_fails_shm()

File: tests/check_ipc.c [hunk 2]
```diff
@@ -429,15 +428,18 @@ run_ipc_server(void)
 		qb_ipcs_enforce_buffer_size(s1, max_size);
 	}
 	qb_ipcs_poll_handlers_set(s1, &ph);
+	qb_loop_signal_add(my_loop, QB_LOOP_HIGH, SIGTERM,
+			   NULL, exit_handler, &handle);
```
Comment (jnpkrn) [inline tests/check_ipc.c]: Good catch that the former location made the very first invocation
of `qb_loop_signal_add` silently fail.

Can we have this fix as a separate commit, please?

File: tests/check_ipc.c [hunk 3]
```diff
@@ -429,15 +428,18 @@ run_ipc_server(void)
...
 	res = qb_ipcs_run(s1);
 	ck_assert_int_eq(res, 0);
 
 	qb_loop_run(my_loop);
+	qb_ipcs_destroy(s1);
 	qb_log(LOG_DEBUG, "loop finished - done ...");
 }
 
-static int32_t
+static pid_t
```
Comment (jnpkrn) [inline tests/check_ipc.c]: Again, good catch, again I'd vote for a separate commit :)

Comment (chrissie-c) [inline tests/check_ipc.c]: Good idea. I'll put those two 'real' fixes in now. The cleanup ones rare being more elusive than I'd like
